### PR TITLE
Feature 1223/create versioned template backend

### PR DESCRIPTION
**Pull request by smithj11**
Go to viewfeedback/responses to view the sample responses in browser.

### CODE REVIEW

**Review by JEdwardBrown — APPROVED** [review]

**Review by moshirim — APPROVED** [review]

File: server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplate.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    @Schema(description = "title of feedback template", required = true)
+    private String title;
+
+    @Column(name = "description")
+    @Nullable
+    @TypeDef(type = DataType.STRING)
+    @Schema(description = "description of feedback template", required = false)
+    private String description;
+
+    @Column(name = "createdBy")
```
Comment (HootDunk) [inline server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplate.java:40]: I know it's been brought up elsewhere but in case it's unnoticed here, camel-case column names aren't working for postgre following the micronaut update.

Comment (pieperm) [reply server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplate.java:40]: I'll also mention that on the refactor in #1167, the `createdBy` column was renamed to `creatorId`. We could do the same thing here for consistency.

**Review by HootDunk — APPROVED** [review]

File: server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplate.java (hunk 1 above)
Comment (smithj11) [reply server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplate.java]: Column names changed to snake case!!

File: server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java [hunk 2]
```diff
@@ -422,18 +410,16 @@ void testGetByCreatorTemplateRequesteeIdMultiplePermitted() {
         createDefaultRole(RoleType.PDL, pdlMemberProfileTwo);
         MemberProfile recipientOne = createADefaultRecipient();
         MemberProfile recipientTwo = createASecondDefaultRecipient();
-
-        UUID identicalTemplateId = UUID.randomUUID();
         LocalDate now = LocalDate.now();
         //create two sample feedback requests by the same PDL and same requestee
-        FeedbackRequest feedbackReq = new FeedbackRequest(UUID.randomUUID(), pdlMemberProfile.getId(), memberOne.getId(), recipientOne.getId(), identicalTemplateId, now, null, "pending", null);
+        FeedbackRequest feedbackReq = new FeedbackRequest(UUID.randomUUID(), pdlMemberProfile.getId(), memberOne.getId(), recipientOne.getId(), now, null, "pending", null);
```
Comment (pieperm) [inline server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java:415]: Rename this test to `testGetByCreatorRequesteeIdMultiplePermitted`?

File: server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java [hunk 3]
```diff
@@ -377,7 +365,7 @@ void testGetByCreatorTemplateRequesteeIdPermitted() {
         createFeedbackRequest(pdlMemberProfileTwo, memberTwo, recipientTwo);
 
         //search for feedback requests by a specific creator, requestee, and template
-        final HttpRequest<?> request = HttpRequest.GET(String.format("/?creatorId=%s&requesteeId=%s&templateId=%s", feedbackReq.getCreatorId(), feedbackReq.getRequesteeId(), feedbackReq.getTemplateId()))
+        final HttpRequest<?> request = HttpRequest.GET(String.format("/?creatorId=%s&requesteeId=%s", feedbackReq.getCreatorId(), feedbackReq.getRequesteeId()))
```
Comment (pieperm) [inline server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java:368]: Rename this test to `testGetByCreatorRequesteeIdPermitted`?

File: server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java [hunk 4]
```diff
@@ -403,7 +391,7 @@ void testGetByCreatorTemplateRequesteeIdNotPermitted() {
...
```
Comment (pieperm) [inline server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java:394]: Rename to `testGetByCreatorRequesteeIdNotPermitted`?

File: web-ui/src/components/template-preview-modal/TemplatePreviewModal.js [hunk 5]
```diff
@@ -1,141 +0,0 @@
...
-    width: '25ch',
-
-  }
-}));
-
-const Transition = React.forwardRef(function Transition(props, ref) {
-  return <Slide direction="up" ref={ref} {...props} />;
-});
-
-const TemplatePreviewModal = ({ open, onSubmit, onClose, template }) => {
```
Comment (pieperm) [inline web-ui/src/components/template-preview-modal/TemplatePreviewModal.js:45]: Why was this file deleted?

**Review by pieperm — CHANGES_REQUESTED** [review]

File: web-ui/src/components/template-preview-modal/TemplatePreviewModal.js (hunk 5 above)
Comment (smithj11) [reply web-ui/src/components/template-preview-modal/TemplatePreviewModal.js]: Fixed, template preview put back

File: server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateController.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                        .headers(headers -> headers.location(URI.create("/frozen_template/" + feedbackTemplate.getId()))))
+                .subscribeOn(Schedulers.from(executorService));
+    }
+
+    private FrozenTemplateResponseDTO fromEntity(FrozenTemplate template) {
+        FrozenTemplateResponseDTO dto = new FrozenTemplateResponseDTO();
+        dto.setId(template.getId());
+        dto.setTitle(template.getTitle());
+        dto.setDescription(template.getDescription());
+        dto.setCreatedBy(template.getCreatedBy());
```
[the remark this answers is not in the document]
Comment (smithj11) [reply server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateController.java]: Fixed

File: server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateController.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        dto.setRequestId(template.getRequestId());
+        return dto;
+    }
+
+    private FrozenTemplate fromDTO(FrozenTemplateCreateDTO dto) {
+       FrozenTemplate template = new FrozenTemplate();
+       template.setCreatedBy(dto.getCreatedBy());
+       template.setTitle(dto.getTitle());
+       template.setDescription(dto.getDescription());
+       template.setCreatedBy(dto.getCreatedBy());
```
[the remark this answers is not in the document]
Comment (smithj11) [reply server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateController.java]: Fixed

File: server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateCreateDTO.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    @Schema(description = "title of feedback template", required = true)
+    private String title;
+
+    @Nullable
+    @Schema(description = "description of feedback template", required = false)
+    private String description;
+
+    @NotBlank
+    @Schema(description = "UUID of person who created the feedback template, not necessarily request creator", required = true)
+    private UUID createdBy;
```
[the remark this answers is not in the document]
Comment (smithj11) [reply server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateCreateDTO.java]: Fixed

File: server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateResponseDTO.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    @Schema(description = "title of feedback template", required = true)
+    private String title;
+
+    @Nullable
+    @Schema(description = "description of feedback template")
+    private String description;
+
+    @NotBlank
+    @Schema(description = "UUID of person who created the feedback template", required = true)
+    private UUID createdBy;
```
[the remark this answers is not in the document]
Comment (smithj11) [reply server/src/main/java/com/objectcomputing/checkins/services/frozen_template/FrozenTemplateResponseDTO.java]: Fixed

File: server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java (hunk 2 above)
Comment (smithj11) [reply server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java:415]: Fixed

File: server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java (hunk 3 above)
Comment (smithj11) [reply server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java:368]: Fixed

File: server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java (hunk 4 above)
Comment (smithj11) [reply server/src/test/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestControllerTest.java:394]: Fixed

**Review by pieperm — APPROVED** [review]
